### PR TITLE
fix(oas): reject invalid bridge timeouts

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
- reject non-positive and non-finite `Masc_oas_bridge.run_safe` timeouts before executing the callback
- document the `Invalid_argument` precondition on the public interface
- add a focused regression test for zero, negative, infinite, and NaN timeouts plus the positive fallback path

## Why it matters
This addresses the Kimi cleanup P0 timeout guard item: invalid bridge budgets should fail deterministically at the boundary instead of flowing into Eio timeout handling or executing OAS work with an impossible budget.

## Validation
- `git diff --check`
- `ocamlformat --check test/test_masc_oas_bridge_timeout_guard.ml`
- `scripts/dune-local.sh build test/test_masc_oas_bridge_timeout_guard.exe`
- `./_build/default/test/test_masc_oas_bridge_timeout_guard.exe` (3 tests passed)
- `scripts/dune-local.sh build test/test_tool_task_coverage.exe`

## Note
`./_build/default/test/test_tool_task_coverage.exe` was run while initially validating the broad placement. The new timeout test passed there, but the binary has an unrelated existing failure in `transition_submit_pr_evidence_accepts_todo_pr_evidence_with_stale_do_not_reclaim_reason` where claim unexpectedly succeeds despite stale `do_not_reclaim_reason`. This PR moved the timeout regression into a focused standalone binary so the bridge guard has a clean verifier.

### CODE REVIEW

File: lib/masc_oas_bridge.mli [hunk 1]
```diff
@@ -7,7 +7,8 @@
     Catches [Eio.Time.Timeout] and [Eio.Cancel.Cancelled] to perform functional rollback.
     [caller] (#10094) labels the Prometheus timeout counter so the
     operator can attribute timeouts to specific call sites; defaults
-    to ["unknown"] for backwards compatibility with legacy callers. *)
+    to ["unknown"] for backwards compatibility with legacy callers.
+    Raises [Invalid_argument] when [timeout_s] is not positive and finite. *)
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/masc_oas_bridge.mli:11]: Addressed in 35f4011145: Env_config_oas_bridge now sanitizes invalid env-derived timeout budgets before run_with_caller delegates to run_safe, and the run_with_caller mli documents the inherited Invalid_argument contract for any bypassed/misconfigured value. Validation: git diff --check passed; attempted scripts/dune-local.sh build test/test_masc_oas_bridge_timeout_guard.exe but stopped my own queued wait because it remained blocked on the shared opam lock.
